app/scripts/directives/serviceIntegration.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('serviceIntegration', {
+    controller: [
+      'APIService',
+      'AuthorizationService',
+      'DataService',
+      'CatalogService',
+      ServiceIntegration
+    ],
+    controllerAs: '$ctrl',
+    bindings: {
+      integration: '<',
+      refApiObject: '<?',
+      serviceClasses: '<',
+      serviceInstances: '<',
+      isOverview: '<?'
+    },
+    templateUrl: 'views/directives/_service-integration.html'
+  });
+
+  function ServiceIntegration(APIService,
+                          AuthorizationService,
+                          DataService,
+                          CatalogService) {
+
+    var ctrl = this;
+    var serviceBindingsResource = {group: "servicecatalog.k8s.io", version: "v1beta1", resource: "servicebindings"};
+    var context = {namespace: ctrl.refApiObject.metadata.namespace};
+    DataService.watch(serviceBindingsResource, context, function(bindingData) {
+      ctrl.binding = false;
+      _.each(bindingData._data, function(binding){
+        if(binding.metadata.annotations.consumer === ctrl.refApiObject.metadata.labels.serviceName){
+          ctrl.binding = binding;
+        }
+      });
+    });
+
+    var serviceInstancesResource = {group: "servicecatalog.k8s.io", version: "v1beta1", resource: "serviceinstances"};
+    DataService.watch(serviceInstancesResource, context, function(serviceInstancesData) {
+      ctrl.serviceInstance = false;
+      _.each(serviceInstancesData._data, function(serviceInstance){
+        if(serviceInstance.metadata.name.includes(ctrl.integration)){
+          ctrl.serviceInstance = serviceInstance;
+        }
+      });
+    });
+
+    CatalogService.getCatalogItems().then(catalogServices => {
+      ctrl.doRender = false;
+      _.each(catalogServices, function(catalogService){
+        if("resource" in catalogService && 
```

Comment:
This doesn't look quite right what are we attempting to do here?

Reply:
yeah, good point, refactoring to use `_.get` instead